hack/update-vendor.sh

```diff
@@ -44,21 +44,23 @@ TMP_DIR="${TMP_DIR:-$(mktemp -d /tmp/update-vendor.XXXX)}"
 LOG_FILE="${LOG_FILE:-${TMP_DIR}/update-vendor.log}"
 kube::log::status "logfile at ${LOG_FILE}"
 
+# Set up some FDs for this script to use, while capturing everything else to
+# the log. NOTHING ELSE should write to $LOG_FILE directly.
+exec 11>&1            # Real stdout, use this explicitly
+exec 22>&2            # Real stderr, use this explicitly
+exec 1>"${LOG_FILE}"  # Automatic stdout
+exec 2>&1             # Automatic stderr
+set -x                # Trace this script to stderr
+
 function finish {
   ret=$?
-  if [[ ${ret} != 0 ]]; then
-    echo "An error has occurred. Please see more details in ${LOG_FILE}"
+  if [[ "${ret}" != 0 ]]; then
```

Comment:
curious about this change... is comparing strings to ints correct? is it ever possible for $? to be undefined?

Reply:
pedantry - habit, should not be in here.  This could just be `$ret` because we know the contents, but habit says to use braces and quotes

Reply:
if it's equivalent, I don't care. I am not fluent enough in bash to know